packages/assets-controllers/src/TokenDetectionController.ts

```diff
@@ -451,16 +451,21 @@ export class TokenDetectionController extends StaticIntervalPollingController<
     }
     const isTokenDetectionInactiveInMainnet =
       !this.#isDetectionEnabledFromPreferences && chainId === ChainId.mainnet;
-    const { tokenList } = this.messagingSystem.call(
+    const { tokensChainsCache } = this.messagingSystem.call(
       'TokenListController:getState',
     );
+    const tokenList = tokensChainsCache[chainId]?.data || {};
+
     const tokenListUsed = isTokenDetectionInactiveInMainnet
       ? STATIC_MAINNET_TOKEN_LIST
       : tokenList;
 
-    const { tokens, detectedTokens, ignoredTokens } = this.messagingSystem.call(
-      'TokensController:getState',
-    );
+    const { allTokens, allDetectedTokens, allIgnoredTokens } =
+      this.messagingSystem.call('TokensController:getState');
+    const tokens = allTokens[chainId]?.[selectedAddress] || [];
+    const detectedTokens = allDetectedTokens[chainId]?.[selectedAddress] || [];
+    const ignoredTokens = allIgnoredTokens[chainId]?.[selectedAddress] || [];
```

Comment:
Nit: Same as above -- prefer `??`

```suggestion
    const tokens = allTokens[chainId]?.[selectedAddress] ?? [];
    const detectedTokens = allDetectedTokens[chainId]?.[selectedAddress] ?? [];
    const ignoredTokens = allIgnoredTokens[chainId]?.[selectedAddress] ?? [];
```

Looks like all four changes slightly lowers the jest coverage threshold.
```diff
--- a/packages/assets-controllers/jest.config.js
+++ b/packages/assets-controllers/jest.config.js
@@ -17,7 +17,7 @@ module.exports = merge(baseConfig, {
   // An object that configures minimum threshold enforcement for coverage results
   coverageThreshold: {
     global: {
-      branches: 88.93,
+      branches: 88.83,
       functions: 96.71,
       lines: 97.34,
       statements: 97.4,
```